actors/organizer.go

```diff
@@ -51,35 +47,59 @@ func (o *Organizer) HandleWholeMessage(msg []byte, userId int) ([]byte, []byte,
 		message, channel, err = nil, nil, handleUnsubscribe(generic, userId)
 	case "publish":
 		message, channel, err = o.handlePublish(generic)
-	//case "message": err = h.handleMessage()
-	//Potentially, we never receive a "message" and only output "message" after a "publish" in order to broadcast.
+	case "message":
+		message, channel, err = o.handleMessage(generic)
 	//Or they are only notification, and we just want to check that it was a success
 	case "catchup":
 		history, err = o.handleCatchup(generic)
 	default:
-		fmt.Printf("2")
+		fmt.Printf("method not recognized, generating default response")
 		message, channel, err = nil, nil, define.ErrRequestDataInvalid
 	}
 
 	return message, channel, define.CreateResponse(err, history, generic)
 }
 
-func handleSubscribe(generic define.Generic, userId int) error {
-	params, err := define.AnalyseParamsLight(generic.Params)
+func (o *Organizer) handleMessage(generic define.Generic) ([]byte, []byte, error) {
+	params, err := define.AnalyseParamsFull(generic.Params)
 	if err != nil {
-		fmt.Printf("3")
-		return define.ErrRequestDataInvalid
+		fmt.Printf("unable to analyse paramsLight in handleMessage()")
```

Comment:
Changed to Params and ParamsIncludingMessage.
It can still be argued that it's not great, but unfortunately, if we want to use clean and nice structure to parse, we have to use two different structures depending on whether params only contains the channel id, vs the channel id + the message content